universe/engine.py

```diff
@@ -1,110 +1,101 @@
-from decimal import Decimal
-import random
+from . import components, systems
+from .orders import (Move, CargoTransfer, Scrap, BuildInstallation, Terraform,
+                     BuildStation, BuildShip, LaunchMassPacket)
 
 
-class GameState(object):
+class Entity:
+    def __init__(self, registry, data):
+        self._data = data
+        self._components = registry[data['type']]
+
+    def __contains__(self, key):
+        return key in self._components
+
+    def serialize(self):
+        data = {'type': self._data['type']}
+        for _type, component in self._components.items():
+            data.update(component.serialize(self._data))
+        return data
+
+
+class Manager:
+    def __init__(self):
+        self._components = {}
+        self._systems = []
+        self._updates = {}
+
+        self._entity_registry = {}
+
+    def register_system(self, system):
+        self._systems.append(system)
+
+    def register_entity_type(self, name, components):
+        if name in self._entity_registry:
+            raise ValueError("{} is already a registered entity type.".format(name))
+        self._entity_registry[name] = {component.name: component for component in components}
+
+    def get_updates(self, _id):
+        return self._updates.get(_id, [])
+
+    def get_components(self, _type):
+        return self._components.get(_type, {})
+
+    def get_component(self, _type, _id):
+        return self._components.get(_type, {}).get(_id)
+
+    def set_component(self, _type, _id, entity):
+        self._components.setdefault(_type, {})[_id] = entity
+
+    def process(self):
+        for system_cls in self._systems:
+            system = system_cls()
+            system.process(self)
+
+    def import_data(self, data, updates):
+        entities = {_id: Entity(self._entity_registry, entity) for _id, entity in (data.get('entities') or {}).items()}
+
+        self._components['position'] = {_id: entity for _id, entity in entities.items() if 'position' in entity}
```

Comment:
Thoughts: Instead of these manual constructions of the component collection dicts, I think we should have a method (`register_entity`?) that does the creation of the `Entity` object, then iterates over all of the components needed by an entity of this type and calls `set_entity` (currently `set_component`) for each component type.